src/librustdoc/clean/mod.rs

```diff
@@ -1644,8 +1683,46 @@ impl Clean<Type> for hir::Ty {
                 FixedVector(box ty.clean(cx), n)
             },
             TyTup(ref tys) => Tuple(tys.clean(cx)),
-            TyPath(None, ref p) => {
-                resolve_type(cx, p.clean(cx), self.id)
+            TyPath(None, ref path) => {
```

Comment:
will this logic work if you have a private type alias that refers to another private type alias?

e.g., `type A = u32; type B = A; fn foo(x: B) { }`?

(Can you also add some tests for that scenario? I could imagine that even if your patch gets it right, it's something future patches might easily break.)